WordPressVIPMinimum/Sniffs/Security/ProperEscapingFunctionSniff.php

```diff
@@ -47,41 +53,45 @@ public function register() {
 	 */
 	public function process_token( $stackPtr ) {
 
-		if ( in_array( $this->tokens[ $stackPtr ]['content'], $this->escaping_functions, true ) === false ) {
+		if ( isset( $this->escaping_functions[ $this->tokens[ $stackPtr ]['content'] ] ) === false ) {
 			return;
 		}
 
 		$function_name = $this->tokens[ $stackPtr ]['content'];
 
-		$echo_or_string_concat = $this->phpcsFile->findPrevious( Tokens::$emptyTokens, $stackPtr - 1, null, true );
+		$data = [ $function_name ];
+
+		$echo_or_concat_or_html = $this->phpcsFile->findPrevious( Tokens::$emptyTokens, $stackPtr - 1, null, true );
 
-		if ( $this->tokens[ $echo_or_string_concat ]['code'] === T_ECHO ) {
+		if ( $this->tokens[ $echo_or_concat_or_html ]['code'] === T_ECHO ) {
 			// Very likely inline HTML with <?php tag.
-			$php_open = $this->phpcsFile->findPrevious( Tokens::$emptyTokens, $echo_or_string_concat - 1, null, true );
+			$php_open = $this->phpcsFile->findPrevious( Tokens::$emptyTokens, $echo_or_concat_or_html - 1, null, true );
 
 			if ( $this->tokens[ $php_open ]['code'] !== T_OPEN_TAG ) {
 				return;
 			}
 
 			$html = $this->phpcsFile->findPrevious( Tokens::$emptyTokens, $php_open - 1, null, true );
 
-			if ( $this->tokens[ $html ]['code'] !== T_INLINE_HTML ) {
+			if ( $this->tokens[ $html ]['code'] === T_INLINE_HTML && $this->is_outside_html_attr_context( $function_name, $this->tokens[ $html ]['content'] ) ) {
+				$message = 'Wrong escaping function, please do not use `%s()` in a context outside of HTML attributes.';
+				$this->phpcsFile->addError( $message, $html, 'notAttrEscAttr', $data );
 				return;
 			}
-		} elseif ( $this->tokens[ $echo_or_string_concat ]['code'] === T_STRING_CONCAT ) {
+		} elseif ( $this->tokens[ $echo_or_concat_or_html ]['code'] === T_STRING_CONCAT || $this->tokens[ $echo_or_concat_or_html ]['code'] === T_COMMA ) {
```

Comment:
* Did you know you can use the WPCS [`Sniff::strip_quotes()`](https://github.com/WordPress/WordPress-Coding-Standards/blob/7da1894633f168fe244afc6de00d141f27517b62/WordPress/Sniff.php#L950) method for more reliably quote stripping ? (will be replaced in WPCS 3.0 by the PHPCSUtils `TextStrings::stripQuotes()` method which does the same)